pulumi/infra/config.py

```
@@ -0,0 +1,62 @@
+
+
+def _sanitize_bucket_component(value: str, label: str) -> str:
+  """Sanitize repo/environment names for S3 bucket compatibility."""
+  candidate = value.strip().lower()
+  candidate = _VALID_CHARS_PATTERN.sub("-", candidate)
+  candidate = _SEQUENTIAL_DOTS.sub(".", candidate)
+  candidate = _SEQUENTIAL_HYPHENS.sub("-", candidate)
+  candidate = _LEADING_TRAILING_NON_ALNUM.sub("", candidate)
+
+  if not candidate:
+    raise ValueError(f"{label} cannot be fully sanitized; please use a different value.")
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)